### DISCUSSION TITLE
Insensitive/sensistive not working for string `eq` filter option?

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#215] [Q&A] [answered]
The filter below doesn't work for me:
```typescript
{
  "filter": {
    "username": { "eq": "atlinx", "mode": "INSENSITIVE"}
  }
}
```
It doesn't match a user that has the name `Atlinx`.

**Comment by edobrb** [comment, marked as answer]
That's because `mode` is used only in StringOperators 
The type is defined as follow:
```typescript
export type StringOperators = {
  contains?: string
  startsWith?: string
  endsWith?: string
  mode?: 'sensitive' | 'insensitive'
}
```

Under the (MongoDb driver) hood the operations contains, startsWith and endsWith use a $regex. The MongoDb $regex support that feature (case sensitive/insensitive) while $eq doesn't. We might use the $regex for `eq` as well when `mode` field is present.
